src/components/Modal/AddMovieForm/index.jsx

```diff
@@ -18,49 +20,48 @@ const AddMovieForm = ({ onSubmit }) => {
     runtime: '',
   };
 
-  const [form, setValue] = useState(defaultState);
+  const formik = useFormik({
+    initialValues: defaultState,
+    validationSchema: MovieFormSchema,
+    onSubmit: (values) => {
+      const data = {
+        ...values,
+        runtime: Number(values.runtime),
+        genres: values.genres.split(', '),
+      };
+      onSubmit(data);
+    },
+  });
 
-  const handleReset = () => setValue(defaultState);
-
-  const handleSubmit = (evt) => {
-    const data = {
-      ...form,
-      runtime: Number(form.runtime),
-      genres: form.genres.split(', '),
-    };
-
-    evt.preventDefault();
-    onSubmit(data);
-  };
-
-  const handleInputChange = () => (evt) => {
-    const { name, value } = evt.target;
-    setValue({ ...form, [name]: value });
-  };
+  const {
+    values, isSubmitting, dirty, isValid, errors, handleSubmit, resetForm, handleChange,
+  } = formik;
 
   return (
     <form
       className="AddMovieForm"
       onSubmit={handleSubmit}
-      onReset={handleReset}
+      onReset={resetForm}
     >
       <Heading>Add Movie</Heading>
       {formField.map(({
-        id, name, label, placeholder,
+        id, name, type, label, placeholder,
       }) => (
         <TextInput
           key={id}
           name={name}
+          type={type}
           label={label}
           placeholder={placeholder}
-          value={form[name]}
+          value={values[name]}
           required
-          onChange={handleInputChange(form[name])}
+          onChange={handleChange}
+          errors={errors[name]}
```

Comment:
It might be better to show errors only when a field was touched all the whole form was submitted. Consider using a `touched` field from formik